cpp/src/parquet/encryption/test_encryption_util.cc

```diff
@@ -0,0 +1,481 @@
+      flba_writer->WriteBatch(1, nullptr, nullptr, &value);
+    }
+  }
+
+  // Close the ParquetFileWriter
+  file_writer->Close();
+
+  return;
+}
+
+void FileDecryptor::DecryptFile(
+    std::string file,
+    std::shared_ptr<FileDecryptionProperties> file_decryption_properties) {
+  std::string exception_msg;
+  parquet::ReaderProperties reader_properties = parquet::default_reader_properties();
+  if (file_decryption_properties) {
+    reader_properties.file_decryption_properties(file_decryption_properties->DeepClone());
+  }
+
+  auto file_reader = parquet::ParquetFileReader::OpenFile(file, false, reader_properties);
+
+  // Get the File MetaData
+  std::shared_ptr<parquet::FileMetaData> file_metadata = file_reader->metadata();
+
+  // Get the number of RowGroups
+  int num_row_groups = file_metadata->num_row_groups();
+
+  // Get the number of Columns
+  int num_columns = file_metadata->num_columns();
+  ASSERT_EQ(num_columns, 8);
+
+  // Iterate over all the RowGroups in the file
+  for (int r = 0; r < num_row_groups; ++r) {
+    // Get the RowGroup Reader
+    std::shared_ptr<parquet::RowGroupReader> row_group_reader = file_reader->RowGroup(r);
+
+    // Get the RowGroupMetaData
+    std::unique_ptr<RowGroupMetaData> rg_metadata = file_metadata->RowGroup(r);
+
+    int64_t values_read = 0;
+    int64_t rows_read = 0;
+    int16_t definition_level;
+    int16_t repetition_level;
+    int i;
+    std::shared_ptr<parquet::ColumnReader> column_reader;
+
+    // Get the Column Reader for the boolean column
+    column_reader = row_group_reader->Column(0);
+    parquet::BoolReader* bool_reader =
+        static_cast<parquet::BoolReader*>(column_reader.get());
+
+    // Get the ColumnChunkMetaData for the boolean column
+    std::unique_ptr<ColumnChunkMetaData> boolean_md = rg_metadata->ColumnChunk(0);
+
+    // Read all the rows in the column
+    i = 0;
+    while (bool_reader->HasNext()) {
+      bool value;
+      // Read one value at a time. The number of rows read is returned. values_read
+      // contains the number of non-null rows
+      rows_read = bool_reader->ReadBatch(1, nullptr, nullptr, &value, &values_read);
+      // Ensure only one value is read
+      ASSERT_EQ(rows_read, 1);
+      // There are no NULL values in the rows written
+      ASSERT_EQ(values_read, 1);
+      // Verify the value written
+      bool expected_value = ((i % 2) == 0) ? true : false;
+      ASSERT_EQ(value, expected_value);
+      i++;
+    }
+    // make sure we got the same number of values the metadata says
+    ASSERT_EQ(boolean_md->num_values(), i);
+
+    // Get the Column Reader for the Int32 column
+    column_reader = row_group_reader->Column(1);
+    parquet::Int32Reader* int32_reader =
+        static_cast<parquet::Int32Reader*>(column_reader.get());
+
+    // Get the ColumnChunkMetaData for the Int32 column
+    std::unique_ptr<ColumnChunkMetaData> int32_md = rg_metadata->ColumnChunk(1);
+
+    // Read all the rows in the column
```

Comment:
There's really a lot of repetition below. I think we could go two ways:
1) you reduce the number of tested types (I don't understand if there's anything different about them: I don't think so, since encryption seems to be a page-level thing)
2) or you find a way to factor out all the writing/reading of example values, such that we don't have so much code to read and maintain when going through the encryption tests

Reply:
This code is not newly written in this pull, I only move here for common use. It's kind of difficult for option 1, since we use this code to read parquet files generated by Java for interop-test, or we have to change the Java unit tests as well.
I will look for some solution for option 2.

Reply:
Ah, sorry, I hadn't noticed it was just moved around. But thanks for looking at it :-)